scripts/license_checker.py

```diff
@@ -131,48 +145,175 @@ def get_target_files(  # pylint: disable=dangerous-default-value
+    "{author}" in the header template.
     """
-    current_year = str(date.today().year)
-    return normalized_text(
-        license_header_template.replace("{year}", current_year).replace(
-            "{author}", author
-        )
-    )
+    return normalized_text(copyright_template.replace("{author}", author))
+
+
+def is_commented_line(line: str, comment_chars: List[str] = COMMENT_CHARS) -> bool:
+    """Checks whether a line is a comment."""
+    line_stripped = line.strip()
+    for commment_char in comment_chars:
+        if line_stripped.startswith(commment_char):
+            return True
+
+    return False
+
+
+def is_empty_line(line: str) -> bool:
+    """Checks whether a line is empty."""
+    return line.strip("\n").strip("\t").strip() == ""
+
+
+def get_header(file_path: Path, comment_chars: List[str] = COMMENT_CHARS):
+    """Extracts the header from a file and normalizes it."""
+    header_lines: List[str] = []
+
+    try:
+        with open(file_path, "r") as file:
+            for line in file:
+                if is_commented_line(
+                    line, comment_chars=comment_chars
+                ) or is_empty_line(line):
+                    header_lines.append(line)
+                else:
+                    break
+    except UnicodeDecodeError as error:
+        raise UnexpectedBinaryFileError(file_path=file_path) from error
+
+    # normalize the lines:
+    header = "".join(header_lines)
+    return normalized_text(header, chars_to_trim=comment_chars)
+
 
+def validate_year_string(year_string: str, min_year: int = MIN_YEAR) -> bool:
+    """Check if the specified year string is valid.
+    Returns `True` if valid or `False` otherwise."""
+    current_year = date.today().year
 
-def check_file_headers(  # pylint: disable=dangerous-default-value
+    # If the year_string is a single number, it must be the current year:
+    if year_string.isnumeric():
+        return int(year_string) == current_year
+
+    # Otherwise, a range (e.g. 2021 - 2022) is expected:
+    match = re.match("(\d+) - (\d+)", year_string)
+
+    if not match:
+        return False
+
+    year_1 = int(match.group(1))
+    year_2 = int(match.group(2))
+
+    # Check the validity of the range:
+    if year_1 >= min_year and year_2 <= year_1:
+        return False
+
+    # year_2 must be equal to the current year:
+    return year_2 == current_year
+
+
+def check_copyright_notice(
+    copyright: str,
+    copyright_template: str = COPYRIGHT_TEMPLATE,
+    author: str = AUTHOR,
+    comment_chars: List[str] = COMMENT_CHARS,
+    min_year: int = MIN_YEAR,
+) -> bool:
+    """Checks the specified copyright text against a template.
+
+    copyright_template (str):
+        A string containing the copyright text to check against the template.
+    copyright_template (str, optional):
+        A string containing a template for the expected license header.
+        You may include "{year}" which will be replace by the current year.
```

Comment:
Other than this typo, we can merge

```suggestion
        You may include "{year}" which will be replaced by the current year.
```